invenio_app_ils/search/api.py

```diff
@@ -222,3 +242,83 @@ def filter_by_keyword_pid(cls, search, keyword_pid, filter_states=None,
             search = search.exclude("terms", state=exclude_states)
 
         return search
+
+
+class DocumentRequestSearch(RecordsSearch):
+    """RecordsSearch for requests."""
+
+    class Meta:
+        """Search only on requests index."""
+
+        index = "document_requests"
+        doc_types = None
+
+    def search_by_document_pid(self, document_pid=None, filter_states=None,
+                               exclude_states=None):
+        """Retrieve requests based on the given document pid."""
+        search = self
+
+        if document_pid:
+            search = search.filter("term", document_pid=document_pid)
+        else:
+            raise MissingRequiredParameterError(
+                description="document_pid is required"
+            )
+
+        if filter_states:
+            search = search.filter("terms", state=filter_states)
+        elif exclude_states:
+            search = search.exclude("terms", state=exclude_states)
+
+        return search
+
+
+def document_request_search_factory(self, search, query_parser=None):
```

Comment:
Maybe re-use `circulation_search_factory`?

Reply:
Moved `invenio_app_ils.circulation.search:circulation_search_factory` to `invenio_app_ils.search.api:filter_by_patron_search_factory` and updated `config.py` so both loans and document requests uses the same search factory.